Tsapko_Homework_3.cpp

```diff
@@ -0,0 +1,178 @@
+    std::cout << "Enter symbols: ";
+    std::cin >> symbols_right;
+    std::cout << "Number of symbols: " << symbols_right.size() << std::endl;
+
+    int len_array{};
+    len_array = symbols_right.size();
+
+    while (len_array != -1) {
+
+        symbols_back += symbols_right[len_array];
+        --len_array;
+    }
+    std::cout << "Reverse string is " << symbols_back << std::endl;
+
+    std::cout << '\n';
+
+
+// PZ_3_2 Sorting items
+
+    constexpr unsigned int iArr_size{ 6 };
+    std::cout << "This program sorting " << iArr_size << " items.\n";
+    int iArr[iArr_size]{};
+    int value{};
+    bool priznak{};
+
+    std::cout << "Enter " << iArr_size << " numbers from range [-500 .. 500]: " << '\n';
+    std::cin >> iArr[0]; //Записываем первый элемент в массив
+
+    if ((iArr[0] < -500) || (iArr[0] > 500)) {
+        std::cout << "You entered wrong number: " << iArr[0] << '\n';
+        return 0;
+    }
+
+    //Цикл приема оставшихся элементов 
+    for (int k{ 1 }; k < iArr_size; k++) {
+
+        std::cin >> value;
+        if ((value < -500) || (value > 500)) {
+            std::cout << "You entered wrong number: " << value << '\n';
+            return 0;
+        }
+
+        /*Изначально все элементы массива были инициализированы нулем, поэтому мы сравниваем введенное
+        значение value только с записанными в массив элементами, т. е. начиная с нулевого элемента массива
+        до номера равного последнему введенному значению value. Оставшиеся элементы массива пока равны нулю.
+        */
+        for (int i{}; i < k; i++) {
+
+            if (iArr[i] > value) {
+
+                priznak = 1; //Признак того, что value меньше одного из элементов массива
+
+                //Это цикл для смещения элементов массива больших, чем значение value
+                for (int t{ iArr_size - 1 }; t > i; t--) {
+
+                    iArr[t] = iArr[t - 1];
+                }
+
+                iArr[i] = value;
+                i = k - 1; //Чтобы не продолжать цикл до конца
+            }
+        }
+
+        /*Если введенное value оказалось больше всех записанных элементов в массив,
+        то присваиваем его следующему элементу массива*/
+        if (priznak != 1) {
+
+            iArr[k] = value;
+
+        }
+
+        priznak = 0; //Обнуляем признак, чтобы опять видеть value больше записанных элементов массива или нет
+    }
+
+    std::cout << '\n';
+
+    for (int i{}; i < iArr_size; i++) {
+
+        std::cout << iArr[i] << '\n';
+    }
+
+    std::cout << '\n';
+
+
+// PZ_3_3 Do upper of entered symbols
+
+    std::cout << "This program do upper of entered symbols." << '\n';
+
+    constexpr int iLow_min{ 96 };
+    constexpr int iLow_max{ 123 };
+    constexpr int iDiff_l_high{ 32 };
+
+    std::string symbols_inp{};
+
+    std::cout << "Enter symbols: ";
+    std::cin >> symbols_inp;
+    std::cout << "Number of symbols: " << symbols_inp.size() << std::endl;
+
+    int len_str{};
+    len_str = symbols_inp.size();
```

Comment:
```suggestion
    int len_str {symbols_inp.size()};
```

выражение `symbols_inp.size()` можно прямо использовать в цикле компилятор умеет такие вещи оптимизировать.
Потом вам компилятор выдаст такое предупреждения что тип size является беззнаковым, а вы присваиваете знаковому, у вас может произойти переполение